Geometry/CSCGeometryBuilder/src/CSCGeometryParsFromDD.cc

```diff
@@ -301,3 +358,272 @@ bool CSCGeometryParsFromDD::build(const DDCompactView* cview,
+
+      for (auto i : fv.get<std::vector<double>>(std::string(namesInPath[index]), "NumWiresPerGrp")) {
+        wg.wiresInEachGroup.emplace_back(int(i));
+      }
+
+      for (auto i : fv.get<std::vector<double>>(std::string(namesInPath[index]), "NumGroups")) {
+        wg.consecutiveGroups.emplace_back(int(i));
+      }
+
+      auto wirespacing = fv.get<double>("WireSpacing");
+      wg.wireSpacing = static_cast<double>(wirespacing);
+
+      auto alignmentpintofirstwire = fv.get<double>("AlignmentPinToFirstWire");
+      wg.alignmentPinToFirstWire = static_cast<double>(alignmentpintofirstwire);
+
+      auto totnumwiregroups = fv.get<double>("TotNumWireGroups");
+      wg.numberOfGroups = static_cast<int>(totnumwiregroups);
+
+      auto lengthoffirstwire = fv.get<double>("LengthOfFirstWire");
+      wg.narrowWidthOfWirePlane = static_cast<double>(lengthoffirstwire);
+
+      auto lengthoflastwire = fv.get<double>("LengthOfLastWire");
+      wg.wideWidthOfWirePlane = static_cast<double>(lengthoflastwire);
+
+      auto radialextentofwireplane = fv.get<double>("RadialExtentOfWirePlane");
+      wg.lengthOfWirePlane = static_cast<double>(radialextentofwireplane);
+
+      uparvals.emplace_back((wg.wireSpacing) * 10.0);
+      uparvals.emplace_back(wg.alignmentPinToFirstWire);
+      uparvals.emplace_back(wg.numberOfGroups);
+      uparvals.emplace_back(wg.narrowWidthOfWirePlane);
+      uparvals.emplace_back(wg.wideWidthOfWirePlane);
+      uparvals.emplace_back(wg.lengthOfWirePlane);
+      uparvals.emplace_back(wg.wiresInEachGroup.size());
+
+      for (CSCWireGroupPackage::Container::const_iterator it = wg.wiresInEachGroup.begin();
+           it != wg.wiresInEachGroup.end();
+           ++it) {
+        uparvals.emplace_back(*it);
+      }
+      for (CSCWireGroupPackage::Container::const_iterator it = wg.consecutiveGroups.begin();
+           it != wg.consecutiveGroups.end();
+           ++it) {
+        uparvals.emplace_back(*it);
+      }
+
+      /** end stuff **/
+    }
+
+    fpar.clear();
+
+    std::string my_title(fv.solid()->GetTitle());
+
+    if (my_title == "Subtraction") {
+      cms::DDSolid mysolid(fv.solid());
+      auto solidA = mysolid.solidA();
+      std::vector<double> dpar = solidA.dimensions();
+
+      std::transform(
+          dpar.begin(), dpar.end(), dpar.begin(), [](double i) -> double { return cms_rounding::roundIfNear0(i); });
+
+      fpar.emplace_back((dpar[1]));
+      fpar.emplace_back((dpar[2]));
+      fpar.emplace_back((dpar[3]));
+      fpar.emplace_back((dpar[4]));
+    } else {
+      dpar = fv.parameters();
+
+      std::transform(
+          dpar.begin(), dpar.end(), dpar.begin(), [](double i) -> double { return cms_rounding::roundIfNear0(i); });
+
+      fpar.emplace_back((dpar[0]));
+      fpar.emplace_back((dpar[1]));
+      fpar.emplace_back((dpar[2]));
+      fpar.emplace_back((dpar[3]));
+    }
+
+    gtran[0] = (float)1.0 * (fv.translation().X());
```

Comment:
```suggestion
    gtran[0] = static_cast<float>(fv.translation().X());
```

Reply:
If you're trying to reduce the precision of the number, `static_cast` is faster. Otherwise, I don't understand why you would multiply by 1.0, other than to waste CPU.
I'm guessing simple assignment would be correct here.
Whatever change you make, it should be made to the following two lines also.

Reply:
Done